include/envoy/network/io_handle.h

```diff
@@ -32,12 +32,16 @@ class IoHandle {
   virtual ~IoHandle() = default;
 
   /**
-   * Return data associated with IoHandle.
+   * NOTE: This is most probably NOT the function you are looking for. IoHandle has wrappers for
+   * most of the POSIX socket api functions so there should be no need to interact with the
+   * internal fd by means of syscalls. Moreover, depending on the IoHandle implementation,
+   * the fd might not be an underlying OS file descriptor. If any api function is missing, a
+   * wrapper for it should be added to the IoHandle interface.
+   *
+   * Return data associated with IoHandle. It is not necessarily a file descriptor.
    *
-   * TODO(danzh) move it to IoSocketHandle after replacing the calls to it with
-   * calls to IoHandle API's everywhere.
    */
-  virtual os_fd_t fd() const PURE;
+  virtual os_fd_t fdInternal() const PURE;
```

Comment:
Can we call this `fdDoNotUse()` and 1) have a note here about not using it for new use cases and 2) can you add a TODO everywhere it is still used with some thoughts on how to eventually clean it up? Thank you!

Reply:
Done but make sure I captured what you were expecting. Also, I'm not sure there's an obvious way to avoid using fds in quic and hot restarting parent, so no comments, for now.